packages/cli/src/controllers/invitation.controller.ts

```diff
@@ -0,0 +1,301 @@
+import validator from 'validator';
+import { In } from 'typeorm';
+import Container, { Service } from 'typedi';
+import { ErrorReporterProxy as ErrorReporter } from 'n8n-workflow';
+import { Authorized, NoAuthRequired, Post, RestController } from '@/decorators';
+import { BadRequestError, InternalServerError, UnauthorizedError } from '@/ResponseHelper';
+import { issueCookie } from '@/auth/jwt';
+import { RESPONSE_ERROR_MESSAGES } from '@/constants';
+import { Response } from 'express';
+import { User } from '@db/entities/User';
+import { UserRequest } from '@/requests';
+import { Config } from '@/config';
+
+import { IExternalHooksClass, IInternalHooksClass } from '@/Interfaces';
+import { License } from '@/License';
+import { UserService } from '@/services/user.service';
+import { Logger } from '@/Logger';
+import { isSamlLicensedAndEnabled } from '@/sso/saml/samlHelpers';
+import {
+	generateUserInviteUrl,
+	getInstanceBaseUrl,
+	hashPassword,
+	validatePassword,
+} from '@/UserManagement/UserManagementHelper';
+import { RoleService } from '@/services/role.service';
+import { UserManagementMailer } from '@/UserManagement/email';
+import { PostHogClient } from '@/posthog';
+
+@Service()
+@RestController('/invitations')
+export class InvitationController {
+	constructor(
+		private readonly config: Config,
+		private readonly logger: Logger,
+		private readonly internalHooks: IInternalHooksClass,
+		private readonly externalHooks: IExternalHooksClass,
+		private readonly userService: UserService,
+		private readonly roleService: RoleService,
+		private readonly mailer: UserManagementMailer,
+		private readonly postHog?: PostHogClient,
```

Comment:
we need to get rid of this optional dependency stuff. then we can use `Container.get`, instead of having to manually instantiate this class everywhere.
If this PR isn't urgent, I'd like to make that change in another PR, before the end of this day.

Reply:
This is not urgent!

Reply:
It think postHog is optional because we do not inject it in the tests for some reason. In we inject it on the test, we can get rid of it